### DISCUSSION TITLE
Debugging large programs: ?

### DISCUSSION THREAD

**Discussion by smurfix** [micropython/micropython#10808] [Ideas]
Hi,

I'm working on some code that's way too large for RAM, so I'm byte-compiling it all to Flash.

However, when there's a problem (or I need to add a `print()` statement …) I want to upload the module I'm working on. Not re-flash the whole thing.

Is there a way to do that?

In a perfect world the `.mpy` files in the file system would be memory-mapped instead of copied to RAM …

**Comment by bixb922** [comment]
Looks like #8381, would be great to have that

**Comment by smurfix** [comment]
While #8381 would be great, frankly it's still uploading a blob (albeit a smaller one) with all my `.mpy` files to a fixed address in Flash. While that would be slightly faster than re-flashing the whole board it does leave concerns about wearing out the flash storage, long- term.

My ideal solution would be for MicroPython to check all elements of the search path for the relevant file-or-directory, not just the first one it finds. Right now, if there's a `/lib/foo/` directory it will no longer find `.flash/foo/bar.mpy` even though `help("modules") does list it and `.flash` still is in `usys.path`.

I filed #10813 as a proposed enhancement to address this.
